### ISSUE TITLE
Take a function string instead of a path/URL

### ISSUE THREAD

**Issue by stevenvachon** [josdejong/workerpool#23]
similar to [function-thread](https://npmjs.com/function-thread).

If you use the `vm` module, it already takes a string. As for `Worker`, you could use a data-url.

**Comment by josdejong** [comment]
Thanks for the suggestion. 

I guess you mean to send a script containing functions instead of passing the url of a script. A single function can already be executed in a worker easily without even having to stringify it ([see docs](https://github.com/josdejong/workerpool#offload-functions-dynamically)).

**Comment by stevenvachon** [comment]
Thanks!

**Comment by josdejong** [comment]
Why do you close the issue? Do you only need passing a single function?

**Comment by stevenvachon** [comment]
The `exec()` method provides the functionality that I was after. I'd just missed it when I skimmed through the readme.

**Comment by josdejong** [comment]
ah ok perfect! let's keep it closed then. 

If anyone is interested in passing a complete script as string we can reopen this issue.

**Comment by stevenvachon** [comment]
Actually, I think I got confused. This is what I'd originally requested:
```js
workerpool.pool("data:text/javascript;utf-8,(function(){…})()")
```

so that we don't need to have a separate "workerpool.js" file

**Comment by josdejong** [comment]
Yes exactly, so you can bundle your worker file in the main file and load it as a string. That can be useful I think.

**Comment by josdejong** [comment]
See also #46 (duplicate)

**Comment by josdejong** [comment]
I've created an example on creating an embedded worker using a data URL, that works out of the box already:

https://github.com/josdejong/workerpool/tree/master/examples/embeddedWorker

Support for data URL's in node.js requires work.

**Comment by imfunniee** [comment]
eagerly waiting for node support 😋
